bindgen/src/bindings/cpp/templates/cpp_scaffolding.cpp

```diff
@@ -0,0 +1,520 @@
+}
+{% endfor %}
+
+{% for checksum in ci.iter_checksums() %}
+uint16_t {{ checksum.0 }}() { return {{ checksum.1 }}; }
+{% endfor %}
+
+{% let contract_fn = ci.ffi_uniffi_contract_version() %}
+uint32_t {{ contract_fn.name() }}() { return {{ ci.uniffi_contract_version() }}; }
+
+#ifdef __cplusplus
+}
+#endif
+
+RustBuffer rustbuffer_alloc(int32_t size) {
+    RustCallStatus status = { 0 };
+
+    return {{ ci.ffi_rustbuffer_alloc().name() }}(
+        size,
+        &status
+    );
+}
+
+RustBuffer rustbuffer_from_bytes(const ForeignBytes& bytes) {
+    RustCallStatus status = { 0 };
+
+    return {{ ci.ffi_rustbuffer_from_bytes().name() }}(
+        bytes,
+        &status
+    );
+}
+
+void rustbuffer_free(RustBuffer& buf) {
+    RustCallStatus status = { 0 };
+
+    {{ ci.ffi_rustbuffer_free().name() }}(
+        buf,
+        &status
+    );
+}
+
+{%- for typ in ci.iter_types() %}
+{%- let type_name = typ|type_name %}
+{%- let ffi_converter_name = typ|ffi_converter_name %}
+{%- let canonical_type_name = typ|canonical_name %}
+{%- let contains_object_references = ci.item_contains_object_references(typ) %}
+
+{%- match typ %}
+{%- when Type::Boolean %}
+{% include "bool_conv.cpp" %}
+{%- when Type::UInt8 %}
+{% include "arith_conv.cpp" %}
+{%- when Type::Int8 %}
+{% include "arith_conv.cpp" %}
+{%- when Type::UInt16 %}
+{% include "arith_conv.cpp" %}
+{%- when Type::Int16 %}
+{% include "arith_conv.cpp" %}
+{%- when Type::UInt32 %}
+{% include "arith_conv.cpp" %}
+{%- when Type::Int32 %}
+{% include "arith_conv.cpp" %}
+{%- when Type::UInt64 %}
+{% include "arith_conv.cpp" %}
+{%- when Type::Int64 %}
+{% include "arith_conv.cpp" %}
+{%- when Type::Float32 %}
+{% include "arith_conv.cpp" %}
+{%- when Type::Float64 %}
+{% include "arith_conv.cpp" %}
+{%- when Type::Bytes %}
+{% include "bytes_conv.cpp" %}
+{%- when Type::Timestamp %}
+{% include "timestamp_helper.cpp" %}
+{%- when Type::Duration %}
+{% include "duration_helper.cpp" %}
+{%- when Type::String %}
+{% include "str_conv.cpp" %}
+{%- when Type::Record { module_path, name } %}
+{% include "rec.cpp" %}
+{%- when Type::Optional { inner_type } %}
+{% include "opt_tmpl.cpp" %}
+{%- when Type::Sequence { inner_type } %}
+{% include "seq_tmpl.cpp" %}
+{%- when Type::Map { key_type, value_type } %}
+{% include "map_tmpl.cpp" %}
+{%- when Type::Enum { module_path, name } %}
+{%- let e = ci|get_enum_definition(name) %}
+{%- if ci.is_name_used_as_error(name) %}
+{%- else %}
+{%- if e.is_flat() %}
+{% include "enum_tmpl.cpp" %}
+{%- endif %}
+{%- endif %}
+{%- when Type::Object { module_path, name, imp } %}
+{%- when Type::CallbackInterface { module_path, name } %}
+{%- let ffi_converter_name = typ|ffi_converter_name %}
+{%- let type_name = typ|type_name %}
+{%- let class_name = type_name|class_name %}
+{%- let canonical_type_name = typ|canonical_name %}
+{%- let iface = ci|get_callback_interface_definition(name) %}
+namespace {
+class {{ iface.name() }}Proxy: public {{ iface.name() }} {
```

Comment:
Is it possible to pull this out into separate template file? Also the same for other blocks of type specific code in this file. This entire file looks really intimidating :sweat_smile:

Reply:
Definitely :D, the code was left inlined to speed up development, now that some tests are present, will move it out :+1: